### ISSUE TITLE
pdnsutil b2bmigrate tinydns to gmysql fails: domain_ids are set to -1

### ISSUE THREAD

**Issue by finferflu** [PowerDNS/pdns#6239]
- Program: pdnsutil 
 - Issue type: Bug report

### Short description

After importing records from a tindydns backend, the `domain_id` for each entry in the records table is set to `-1`.

### Environment

 - Operating system: CentOS 7
 - Software version: 4.1.0
 - Software source: PowerDNS repository 

### Steps to reproduce

1. Set up PDNS database as indicated [here](https://doc.powerdns.com/md/authoritative/backend-generic-mysql/#default-schema). 
2. Follow the instructions [here](https://doc.powerdns.com/md/authoritative/migration/#moving-from-source-to-target) up to step 4.
3. `pdnsutil check-all-zones` yields `[error] No SOA record present, or active, in zone 'example.com'` for each zone.
4. Verify that `domain_id` for each record in the records table is set to `-1`.

### Expected behaviour

`records.domain_id` to be set to the ID of the zone in the `domains` table.

### Actual behaviour

`records.domain_id` is set to `-1` for each record.

### Other information

I have also subsequently created constraints as indicated [here](https://doc.powerdns.com/md/authoritative/backend-generic-mysql/#generic-mysql-backend), and running `pdnsutil -v b2b-migrate tinydns gmysql` yields (quite predictably):

```
Processing 'example.com'
Error: GSQLBackend unable to feed record: Could not execute mysql statement: insert into records (content,ttl,prio,type,domain_id,disabled,name,ordername,auth,change_date) values (?,?,?,?,?,?,?,?,?,NULL): Cannot add or update a child row: a foreign key constraint fails (`pdns`.`records`, CONSTRAINT `records_domain_id_ibfk` FOREIGN KEY (`domain_id`) REFERENCES `domains` (`id`) ON DELETE CASCADE ON UPDATE CASCADE)
```

I'd also just like to add that there's nothing fancy in my pdns.conf file, just credentials and backends:

```
# cat /etc/pdns/pdns.conf
launch=tinydns,gmysql
tinydns-dbfile=/root/data.cdb

gmysql-dbname=pdns
gmysql-host=127.0.0.1
gmysql-port=3306
gmysql-user=pdns
gmysql-password=password

version-string=powerdns
```

Please let me know if you'd like any more details, and I'll be happy to provide them.

Thank you!

**Comment by Habbie** [comment]
Hello! Can you test #6559? We can build packages if you like.

**Comment by finferflu** [comment]
Thanks, it'd be useful if you could package it for CentOS 7. Thank you.

**Comment by Habbie** [comment]
Build started at https://builder.powerdns.com/#/builders/8/builds/2264

In about an hour, you should be able to click through to the centos-7 build and get a tarball full of packages from there.

**Comment by mnordhoff** [comment]
builder.powerdns.com only has newer files, but you can get it from https://downloads.powerdns.com/autobuilt/auth/0.0.dinew.2264g604e9ac/pdns-0.0.dinew.2264g604e9ac-centos-7-x86_64.tar.bz2

**Comment by finferflu** [comment]
@mnordhoff Thanks for the blazing fast reply! I'm currently testing these and will report back. Thanks again.

**Comment by finferflu** [comment]
I can confirm that the issue is now fixed. Thank you very much for providing a fix.

**Comment by MANDANGA** [comment]
Hello,

Is possible to apply this fix to the 4.1.2?

regards,
